src/stores/widgets/SpectralProfileWidgetStore.ts

```diff
@@ -193,7 +207,10 @@ export class SpectralProfileWidgetStore extends RegionWidgetStore {
                 let spectralConfig = regionRequirements.spectralProfiles.find(profiles => profiles.coordinate === coordinate);
                 if (!spectralConfig) {
                     // create new spectral config
-                    regionRequirements.spectralProfiles.push({coordinate, statsTypes: [statsType]});
+                    // Qi, creaet new spectral config according coordinate
+                    this.requiredCoordinate(coordinate).forEach(data => {
```

Comment:
This logic is not quite right: At the moment, if the coordinate is `PIz`, then `spectralConfig` will always be false. We then insert a new entry for both `Qz` and `Uz`. This means that we can end up with duplicate entries for `Qz` and `Uz`.

Instead, you should put the `SpectralProfileWidgetStore.requiredCoordinates(coordinate).foreach` around the entire requirements `find` section. That way, you'll check for the exisitance of `Qz` and `Uz` individually.